addons/api/package.json

```diff
@@ -69,9 +69,6 @@
     "qunit-dom": "^2.0.0",
     "webpack": "^5.78.0"
   },
-  "peerDependencies": {
```

Comment:
Why were peer dependencies removed? Don't we want them?

Reply:
Peer dependencies shouldn't be removed from package.json. Also, the dependencies declared within peerDependencies, in this case `ember-source` need to be installed at developer responsability.

In fact (**not to be resolved within this PR**) currently, yarn is complaining about `ember-source` been umet as peer dependency within api, core and rose, I attach screenshot of my terminal after running `yarn install`

![Screenshot 2023-11-10 at 1 21 28 PM](https://github.com/hashicorp/boundary-ui/assets/9775006/eb75cb1e-f8ba-4353-b234-038e84ddb3a6)

Reply:
@ZedLi I guess a question of learning would be why would we need a peer-dependency for Ember source 4.0.0 when we have it already for addons and ui?